gcp/appengine/Dockerfile

```diff
@@ -0,0 +1,54 @@
+# Build the Javascript frontend
+FROM node:latest AS FRONTEND3_BUILD
+WORKDIR /build/frontend3
+
+# Install dependencies first for better caching
+COPY gcp/appengine/frontend3/package.json gcp/appengine/frontend3/package-lock.json ./
+RUN npm ci
+
+COPY gcp/appengine/frontend3/webpack.prod.js ./
+COPY gcp/appengine/frontend3/img img
+COPY gcp/appengine/frontend3/src src
+
+RUN npm run build:prod
+
+# Build hugo blogs
+FROM golang:1.20 AS HUGO_BUILD
+# Build hugo from source - this should usually be cached.
+RUN go install -tags extended github.com/gohugoio/hugo@v0.111.3
+
+WORKDIR /build/blog
+COPY gcp/appengine/blog ./
+
+RUN hugo -d ../dist/static/blog
+
+# OSV.dev site image
+# Adapted from https://cloud.google.com/run/docs/quickstarts/build-and-deploy/deploy-python-service#writing
+FROM python:3.11-slim
+
+# Allow statements and log messages to immediately appear in the logs
+ENV PYTHONUNBUFFERED True
+WORKDIR /osv/gcp/appengine
+
+# Install Python dependencies
+COPY setup.py Pipfile* README.md /osv/
+COPY osv /osv/osv
+COPY gcp/appengine/Pipfile* ./
+RUN pip3 install -U pipenv==2023.6.12 && python3 -m pipenv install --deploy --system
+
+# Website Python code
+COPY gcp/appengine/*.py ./
+
+# JS/hugo builds
+COPY gcp/appengine/dist/public_keys dist/public_keys
+COPY gcp/appengine/docs docs
+COPY --from=FRONTEND3_BUILD /build/dist/ dist/
+COPY --from=HUGO_BUILD /build/dist dist/
+
+
+# Run the web service on container startup. Here we use the gunicorn
```

Comment:
Could you speak (either here, or at next week's team meeting) to your choice of gunicorn? 

I'm mostly interested to understand how much due diligence has gone into the selection (I'm thinking through the lens of https://best.openssf.org/Concise-Guide-for-Evaluating-Open-Source-Software). 

While this is an interim measure to moving to Cloud Run, I imagine this particular aspect of things won't necessarily be temporary, so I'd like to ensure we're being mindful of the technical debt we're taking on.

Reply:
I only chose gunicorn because App Engine standard uses it under the hood for our current website. (And also because it's used in [this quickstart](https://cloud.google.com/run/docs/quickstarts/build-and-deploy/deploy-python-service#writing))

I hadn't put much thought into it otherwise.